Update and checking for DnD5E v2.1.2 + some new feature

p4535992 opened the pull request:
Many people were "unhappy" about my choice of colors for spell types, so to keep everyone happy and cut corners I added the ability to customize colors from the settings.

Other people, on the other hand, want colors , but only for certain preparation types. For example for warlock they don't want to have the colors because it doesn't make sense, I also added this option, just set the value #00000000 to the input color.

Also some people asked me if I could add a reset setting button for the module so i added that too.

I think someone more skilled than me can make a better code ...

**Summary**
- New feature: Added the ability to customize colors 
- New feature: A reset setting button
- Upgrade to dnd5e system version 2.1.2 (ty to the intial PR https://github.com/sdenec/tidy5e-sheet/pull/630 for the intial hint)

There are problems , but so far they all seem to be related to the system upgrade.

I touch a lot of stuff so do some test  before create the release.

Issues related:

https://github.com/sdenec/tidy5e-sheet/issues/629
https://github.com/sdenec/tidy5e-sheet/issues/628
https://github.com/sdenec/tidy5e-sheet/issues/627
https://github.com/sdenec/tidy5e-sheet/issues/625
https://github.com/sdenec/tidy5e-sheet/issues/624
https://github.com/sdenec/tidy5e-sheet/issues/621

MrPrimate commented:
@p4535992 would it be better to put the migration in an `init` or `ready` world hook, check a setting to see if it's been run, then loop over the world actors? The migration will need to be called only once before moving to 1.2.x, so doesn't really make sense to be calling that block every time we open a sheet, and also, folk probably don't want to open each sheet to migrate all the actors in their world.

szefo09 commented:
This one-liner could fast-migrate all the data. It's a dirty method though, as it will add this flag to every character.
```js
//Update old tidy5e data
game.actors.updateAll((actor)=>{
    return {
        "flags.tidy5e-sheet.details":{
            gender:actor.system.details.gender,
            age:actor.system.details.age,
            height:actor.system.details.height,
            weight:actor.system.details.weight,
            eyes:actor.system.details.eyes,
            skin:actor.system.details.skin,
            hair:actor.system.details.hair,
            notes1name:actor.system.details.notes1name,
            notes1:actor.system.details.notes1,
            notes2name:actor.system.details.notes2name,
            notes2:actor.system.details.notes2,
            notes3name:actor.system.details.notes3name,
            notes3:actor.system.details.notes3,
            notes4name:actor.system.details.notes4name,
            notes4:actor.system.details.notes4
        }
    }
})
```

p4535992 commented:
thanks @szefo09  just a small modification on your macro based on the structure I choose (we can always change it ), tell me if it makes more or less sense to you:

```
//Update old tidy5e data
game.actors.updateAll((actor)=>{
    return {
        "flags.tidy5e-sheet":{
            gender:actor.system.details.gender,
            age:actor.system.details.age,
            height:actor.system.details.height,
            weight:actor.system.details.weight,
            eyes:actor.system.details.eyes,
            skin:actor.system.details.skin,
            hair:actor.system.details.hair,
	    notes: {
			 "value": actor.system.details.notes
	    },
            notes1:{
			"name": actor.system.details.notes1name,
			"value": actor.system.details.notes1.value
	    },
	    notes2:{
			"name": actor.system.details.notes2name,
			"value": actor.system.details.notes2.value
	   },
	   notes3:{
			"name": actor.system.details.notes3name,
			"value": actor.system.details.notes3.value
	    },
            notes4:{
			"name": actor.system.details.notes4name
			"value": actor.system.details.notes4.value
	    }
        }
    }
})
```

szefo09 commented:
yeah, this will also work. It could also be split into something like this for every detail:
```js
game.actors.updateAll(
(actor)=> { return {"flags.tidy5e-sheet":{ gender:actor.system.details.gender}}},
(actor)=> { return actor.system.details}
)
```
this second part is a condition if the value exists, as it will return false if it's undefined or null, to not add that parameter.
https://foundryvtt.com/api/classes/client.Actors.html#updateAll
But I'm not sure if it's actually any better than just importing everything as done in the one-liner.

szefo09 commented:
> thanks @szefo09 just a small modification on your macro based on the structure I choose (we can always change it ), tell me if it makes more or less sense to you

I would also follow @MrPrimate advice and made it so that this code runs once befor migration, and then sets a flag that it was done, similar to how it's done in system migration.

p4535992 commented:
does it make sense in your opinion to ask to add a `preHookMigration` at the system level ? because I would like to be 100% sure that the module code is invoked at the right time.

MrPrimate commented:
> `preHookMigration`

Does this exist already, or is it something you'd like adding? If it doesn't exist, then a request sounds like a good idea for the future, but I don't think we can relly on it for this PR.

sdenec commented:
Do you guys drop me a message when you deem it save to merge and release your fixes?

p4535992 commented:
> Do you guys drop me a message when you deem it save to merge and release your fixes?

yes of course

p4535992 commented:
At this point before the official release there would be to implement the discussed "migration tool", but I need to think about how to implement it in a smart way for now i have prepared a utility method : 
https://github.com/p4535992/tidy5e-sheet/blob/master/src/scripts/app/migration-util.js

@MrPrimate  let me know if you want to take care of the migration behaviour because you are more "smart" than me and this is probably the most delicate part.

For anyone interested or just can't wait here the link to the beta:

 https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.0/module.json

Please test it first in a test environment before using it in production.

MrPrimate commented:
@p4535992 I'll PR some changes for the migration against this branch

p4535992 commented:
it's seem i have broken skills and tools i will check out

p4535992 commented:
New link to the beta: https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.1/module.json

p4535992 commented:
![](https://user-images.githubusercontent.com/30846057/213329890-fed082cb-bab8-4c89-9dbe-0db562f5b95a.png)

we will probably have to "remove" the tidy contextmenu and "enrich" the basic system contextmenu (we would over apply the tidy style to the system context menu in my opinion)

nirurin commented:
So I'm trying out the beta, and it certainly seems to fix a lot of issues! 

One issue I haven't seen mentioned though, and which is still broken - favourites. 

Some characters have them, some don't. 
Before the beta, some people could add and remove favourites (though the menu was janky because there was the default menu and the tidy menu overlapping). But now it seems the Tidy menu has been removed, so there's no way to add or remove favourites. 

Just wanted to point it out, as I hadn't seen it mentioned elsewhere, but favourites are -hugely- helpful for players!

SyntOwl commented:
At the footer of the spellbook, the characteristic does not appear.

Change from: `{{#each system.abilities as |abl a|}}`
to: `{{#each abilities as |abl a|}}`

https://github.com/p4535992/tidy5e-sheet/blob/c0fc8166fb5134bb30de59ec12ce368fcdac25b3/src/templates/actors/parts/tidy5e-spellbook-footer.html#L12-L14

p4535992 commented:
New Beta : https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.3/module.json

- Fixed (maybe): https://github.com/sdenec/tidy5e-sheet/issues/633
- Fixed missing spellchooll label
- Removed the old context menu and replace with a new integration with the new ContextMenu of the system
- Fixed abilities label on the spellbook footer
- Rewrite the favorite code for better compatibility with the sistem and should be solve https://github.com/sdenec/tidy5e-sheet/issues/636, https://github.com/sdenec/tidy5e-sheet/issues/634, https://github.com/sdenec/tidy5e-sheet/issues/632

**Issue  i cannot solved**

When the module settings form "Use classic item controls in list view." is enabled it shows the actions correctly but the section headers are "not synchronized at the column level" should be fixed with CSS ...

![](https://i.imgur.com/UoIz6Y1.png)

AliciaBytes commented:
@p4535992 I tried installing the new beta, but getting an error for an invalid path argument:

![image](https://user-images.githubusercontent.com/68741472/213924678-4ebe1c90-2de8-4367-8ba3-40948cb7d576.png)

szefo09 commented:
Did you figure out the notes migration?

p4535992 commented:
> @p4535992 I tried installing the new beta, but getting an error for an invalid path argument:
> 
> ![image](https://user-images.githubusercontent.com/68741472/213924678-4ebe1c90-2de8-4367-8ba3-40948cb7d576.png)

try out again

p4535992 commented:
> Did you figure out the notes migration?

The only way to do that the "right" way is with a preHook... https://github.com/foundryvtt/dnd5e/issues/2068

MrPrimate commented:
@szefo09 @p4535992  after talking to some foundry staff and trying somethings, migration will only be possible by running the macros in a version prior to 2.1.x

szefo09 commented:
Hey, at least now we know. I would suggest releasing an update for tidy5e for DnD5e 2.0.3 which runs the conversion automatically and a version for 2.1.X that doesn’t. This way people that stayed on 2.0.3 could preserve their data? Or is it too complicated and it’s simpler to just warn users in changelog?

p4535992 commented:
FOR ANYONE STILL HAVING THE 2.0.3 BACKUP HERE THE "MIGRATION MACRO" FOR 2.1.X: 

DO A BACKUP BEFORE LAUNCH THIS JUST TO BE SAFE

```
//Update old tidy5e data
game.actors.updateAll((actor)=>{
    return {
        "flags.tidy5e-sheet":{
            gender:actor.system.details?.gender ?? "",
            age:actor.system.details?.age ?? "",
            height:actor.system.details?.height ?? "",
            weight:actor.system.details?.weight ?? "",
            eyes:actor.system.details?.eyes ?? "",
            skin:actor.system.details?.skin ?? "",
            hair:actor.system.details?.hair ?? "",
	    notes: {
			 "value": actor.system.details?.notes  ?? ""
	    },
            notes1:{
			"name": actor.system.details.notes1name ?? "",
			"value": actor.system.details?.notes1  ?? ""
	    },
	    notes2:{
			"name": actor.system.details.notes2name ?? "",
			"value": actor.system.details?.notes2  ?? ""
	   },
	   notes3:{
			"name": actor.system.details.notes3name  ?? "",
			"value": actor.system.details?.notes3  ?? ""
	    },
            notes4:{
			"name": actor.system.details.notes4name  ?? "",
			"value": actor.system.details?.notes4  ?? ""
	    }
        }
    }
})
```

xeropresence commented:
Is the 0.8.3 version not using that data? 
I ran the macro, I can see the populated tidy5e-sheet field in the exported json, but when I import it into a new world running 2.1.2 and 0.8.3 the fields are still empty.

p4535992 commented:
> I ran the macro, I can see the populated tidy5e-sheet field in the exported json, but when I import it into a new world running 2.1.2 and 0.8.3 the fields are still empty.

export the actor and pass to me maybe i miss something

p4535992 commented:
ty to @szefo09 this shoudl be the correct macro

FOR ANYONE STILL HAVING THE 2.0.3 BACKUP HERE THE "MIGRATION MACRO" FOR 2.1.X: 

DO A BACKUP BEFORE LAUNCH THIS JUST TO BE SAFE

```
//Update old tidy5e data
game.actors.updateAll((actor)=>{
    return {
        "flags.tidy5e-sheet":{
            gender:actor.system.details?.gender ?? "",
            age:actor.system.details?.age ?? "",
            height:actor.system.details?.height ?? "",
            weight:actor.system.details?.weight ?? "",
            eyes:actor.system.details?.eyes ?? "",
            skin:actor.system.details?.skin ?? "",
            hair:actor.system.details?.hair ?? "",
	    notes: {
			 "value": actor.system.details?.notes?.value  ?? ""
	    },
            notes1:{
			"name": actor.system.details?.notes1name ?? "",
			"value": actor.system.details?.notes1  ?? ""
	    },
	    notes2:{
			"name": actor.system.details?.notes2name ?? "",
			"value": actor.system.details?.notes2  ?? ""
	   },
	   notes3:{
			"name": actor.system.details?.notes3name  ?? "",
			"value": actor.system.details?.notes3  ?? ""
	    },
            notes4:{
			"name": actor.system.details?.notes4name  ?? "",
			"value": actor.system.details?.notes4  ?? ""
	    }
        }
    }
})
```

p4535992 commented:
New beta: https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.4/module.json

szefo09 commented:
I think you messed up the notes1 to notes4 value as well
![image](https://user-images.githubusercontent.com/37833135/214175808-36bb5607-3073-48e6-aa8f-42d3c5a672e5.png)

p4535992 commented:
ty to @szefo09 this shoudl be the correct macro

FOR ANYONE STILL HAVING THE 2.0.3 BACKUP HERE THE "MIGRATION MACRO" FOR 2.1.X: 

DO A BACKUP BEFORE LAUNCH THIS JUST TO BE SAFE

```
//Update old tidy5e data
game.actors.updateAll((actor)=>{
    return {
        "flags.tidy5e-sheet":{
            gender:actor.system.details?.gender ?? "",
            age:actor.system.details?.age ?? "",
            height:actor.system.details?.height ?? "",
            weight:actor.system.details?.weight ?? "",
            eyes:actor.system.details?.eyes ?? "",
            skin:actor.system.details?.skin ?? "",
            hair:actor.system.details?.hair ?? "",
	    notes: {
			 "value": actor.system.details?.notes?.value  ?? ""
	    },
            notes1:{
			"name": actor.system.details?.notes1name ?? "",
			"value": actor.system.details?.notes1?.value  ?? ""
	    },
	    notes2:{
			"name": actor.system.details?.notes2name ?? "",
			"value": actor.system.details?.notes2?.value  ?? ""
	   },
	   notes3:{
			"name": actor.system.details?.notes3name  ?? "",
			"value": actor.system.details?.notes3?.value  ?? ""
	    },
            notes4:{
			"name": actor.system.details?.notes4name  ?? "",
			"value": actor.system.details?.notes4?.value  ?? ""
	    }
        }
    }
})
```

szefo09 commented:
https://github.com/p4535992/tidy5e-sheet/blob/cce7208f83452a6b889562683ad7070c9da8c8c1/src/templates/actors/parts/tidy5e-inventory.html#L151-L153

`item.totalWeight` to `item.system.weight`
`item.system.price` to `item.system.price.value`
`localize 'DND5E.CurrencyAbbrGP'` to `item.system.price.denomination`
Also, I've DM'd you on Discord about some other fixes.

p4535992 commented:
ty to @szefo09 for point out some issue
NEW BETA: https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.6/module.json

p4535992 commented:
NEW BETA: https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.7/module.json

szefo09 commented on src/scripts/app/context-menu.js:
```diff
@@ -347,6 +352,17 @@ const _getItemContextOptions = function(item) {
+          }
+      });
+    }
+  }
```
Let the Let's Trade 5e module handle it, don't need to add integration this way.

szefo09 replied:
![image](https://user-images.githubusercontent.com/37833135/214440146-ae320c23-c80c-4401-b317-95055d7e613e.png)
Otherwise we will have duplicates like these.

CaosFR commented:
Thanks for the update Sadly the journal of all sheet it's wipe. I have export a backup of sheet but soon have launch the game they are wipe.

nivthefox commented:
Using the latest Beta, when I try to update an NPC's Biography-tab entries (Traits, Flaws, etc.), it wipes the contents of the entire tab.  

```commons.js:1703 Error: You are accessing the "data" attribute within the rendering context provided by the ActorSheet5e 
        class. This attribute has been deprecated in favor of "system" and will be removed in a future release
Deprecated since Version DnD5e 2.0
Backwards-compatible support will be removed in Version DnD5e 2.2
    at Object.logCompatibilityWarning (commons.js:1692:19)
    at Object.get (base-sheet.mjs:119:23)
    at NpcSheetCurrency._getTidyTemplate (npc-sheet-currency.js:85:26)
    at NpcSheetCurrency._handleTidyInjection (npc-sheet-currency.js:69:34)
    at NpcSheetCurrency.injectCurrencyRow (npc-sheet-currency.js:24:18)
    at Object.fn (main.js:28:26)
    at #call (foundry.js:724:20)
    at Hooks.callAll (foundry.js:681:17)
    at Tidy5eNPC._render (foundry.js:5153:13)
    at async Tidy5eNPC._render (foundry.js:5822:5)```

szefo09 commented:
> Using the latest Beta, when I try to update an NPC's Biography-tab entries (Traits, Flaws, etc.), it wipes the contents of the entire tab.


Good catch. @p4535992 has not updated the NPC sheet properly.

Weird case, as NPCs don't have traits, ideals, bonds and flaws by default, but PC Actors do, so Tidy5e has to store NPC traits/ideals/bonds/flaws as a flag.

https://github.com/p4535992/tidy5e-sheet/blob/9fc6922e1c8844335f45bfe4f2763a7676a38f75/src/templates/actors/tidy5e-npc.html#L361-L367
change to:
```html
              <li><span>{{ localize "TIDY5E.Gender" }}:</span><span {{#if owner}}contenteditable="true"{{/if}} spellcheck="false" data-target="{{actor._id}}-gender" data-maxlength="40">{{actor.flags.tidy5e-sheet.gender}}</span><input data-input="{{actor._id}}-gender" type="hidden" name="flags.tidy5e-sheet.gender" value="{{actor.flags.tidy5e-sheet.gender}}"></li>
              <li><span>{{ localize "TIDY5E.Age" }}:</span><span {{#if owner}}contenteditable="true"{{/if}} spellcheck="false" data-target="{{actor._id}}-age" data-maxlength="5">{{actor.flags.tidy5e-sheet.age}}</span><input data-input="{{actor._id}}-age" type="hidden" name="flags.tidy5e-sheet.age" value="{{actor.flags.tidy5e-sheet.age}}"></li>
              <li><span>{{ localize "TIDY5E.Height" }}:</span><span {{#if owner}}contenteditable="true"{{/if}} spellcheck="false" data-target="{{actor._id}}-height" data-maxlength="20">{{actor.flags.tidy5e-sheet.height}}</span><input data-input="{{actor._id}}-height" type="hidden" name="flags.tidy5e-sheet.height" value="{{actor.flags.tidy5e-sheet.height}}"></li>
              <li><span>{{ localize "TIDY5E.Weight" }}:</span><span {{#if owner}}contenteditable="true"{{/if}} spellcheck="false" data-target="{{actor._id}}-weight" data-maxlength="20">{{actor.flags.tidy5e-sheet.weight}}</span><input data-input="{{actor._id}}-weight" type="hidden" name="flags.tidy5e-sheet.weight" value="{{actor.flags.tidy5e-sheet.weight}}"></li>
              <li><span>{{ localize "TIDY5E.Eyes" }}:</span><span {{#if owner}}contenteditable="true"{{/if}} spellcheck="false" data-target="{{actor._id}}-eyes" data-maxlength="40">{{actor.flags.tidy5e-sheet.eyes}}</span><input data-input="{{actor._id}}-eyes" type="hidden" name="flags.tidy5e-sheet.eyes" value="{{actor.flags.tidy5e-sheet.eyes}}"></li>
              <li><span>{{ localize "TIDY5E.Skin" }}:</span><span {{#if owner}}contenteditable="true"{{/if}} spellcheck="false" data-target="{{actor._id}}-skin" data-maxlength="40">{{actor.flags.tidy5e-sheet.skin}}</span><input data-input="{{actor._id}}-skin" type="hidden" name="flags.tidy5e-sheet.skin" value="{{actor.flags.tidy5e-sheet.skin}}"></li>
              <li><span>{{ localize "TIDY5E.Hair" }}:</span><span {{#if owner}}contenteditable="true"{{/if}} spellcheck="false" data-target="{{actor._id}}-hair" data-maxlength="40">{{actor.flags.tidy5e-sheet.hair}}</span><input data-input="{{actor._id}}-hair" type="hidden" name="flags.tidy5e-sheet.hair" value="{{actor.flags.tidy5e-sheet.hair}}"></li>
```
and 
https://github.com/p4535992/tidy5e-sheet/blob/9fc6922e1c8844335f45bfe4f2763a7676a38f75/src/templates/actors/tidy5e-npc.html#L374-L387
change to:
```html
						<div class="section-titles biopage">{{ localize "DND5E.PersonalityTraits" }}</div>
						{{editor actor.flags.tidy5e-sheet.trait target="flags.tidy5e-sheet.trait" button=true editable=editable engine="prosemirror" collaborate=false}}
					</article>
					<article>
						<div class="section-titles biopage">{{ localize "DND5E.Ideals" }}</div>
						{{editor actor.flags.tidy5e-sheet.ideal target="flags.tidy5e-sheet.ideal" button=true editable=editable engine="prosemirror" collaborate=false}}
					</article>
					<article>
						<div class="section-titles biopage">{{ localize "DND5E.Bonds" }}</div>
						{{editor actor.flags.tidy5e-sheet.bond target="flags.tidy5e-sheet.bond" button=true editable=editable engine="prosemirror" collaborate=false}}
					</article>
					<article>
						<div class="section-titles biopage">{{ localize "DND5E.Flaws" }}</div>
						{{editor actor.flags.tidy5e-sheet.flaw target="flags.tidy5e-sheet.flaw" button=true editable=editable engine="prosemirror" collaborate=false}}
```

p4535992 commented:
I add NPC and character and avoid to check vehicle type for avoid to set not wanted flags.

FOR ANYONE STILL HAVING THE 2.0.3 BACKUP HERE THE "MIGRATION MACRO" FOR 2.1.X:

DO A BACKUP BEFORE LAUNCH THIS JUST TO BE SAFE

```
//Update old tidy5e CHARACTER data
game.actors.updateAll((actor)=>{
    return {
        "flags.tidy5e-sheet":{
            gender:actor.system.details?.gender ?? "",
            age:actor.system.details?.age ?? "",
            height:actor.system.details?.height ?? "",
            weight:actor.system.details?.weight ?? "",
            eyes:actor.system.details?.eyes ?? "",
            skin:actor.system.details?.skin ?? "",
            hair:actor.system.details?.hair ?? "",
	    notes: {
			 "value": actor.system.details?.notes?.value  ?? ""
	    },
            notes1:{
			"name": actor.system.details?.notes1name ?? "",
			"value": actor.system.details?.notes1?.value  ?? ""
	    },
	    notes2:{
			"name": actor.system.details?.notes2name ?? "",
			"value": actor.system.details?.notes2?.value  ?? ""
	   },
	   notes3:{
			"name": actor.system.details?.notes3name  ?? "",
			"value": actor.system.details?.notes3?.value  ?? ""
	    },
            notes4:{
			"name": actor.system.details?.notes4name  ?? "",
			"value": actor.system.details?.notes4?.value  ?? ""
	    }
        }
    }
}, (actor)=>{
	return actor.type === 'character';
});

//Update old tidy5e NPC Traits data
game.actors.updateAll((actor)=>{
    return {
        "flags.tidy5e-sheet":{
            trait:actor.system.details?.trait ?? "",
            ideal:actor.system.details?.ideal ?? "",
            bond:actor.system.details?.bond ?? "",
            flaw:actor.system.details?.flaw ?? ""
        }
    }
}, (actor)=>{
	return actor.type === 'npc';
});
```

p4535992 commented:
> Thanks for the update Sadly the journal of all sheet is wipe. I have export a backup of character sheet but when I launch the game they are wipe. Can you do something ?

Can you pass to me a actor both  the pre and post  version ? you can pass to me the exported files on discord

CaosFR commented:
@p4535992 Thanks for your help, where I can find your discord ?

p4535992 commented:
> @p4535992 Thanks for your help, where I can find your discord ?

just PM my user _4535992#1766_

CaosFR commented:
@p4535992 I seed it. Gender, Age, Height, Weight, Eyes, Skin and Hair are also wipe.

In JSON file this colloms seens dont keep :

      "gender": "",
      "age": "",
      "height": "",
      "weight": "",
      "eyes": "",
      "skin": "",
      "hair": "",
      "notes1name": "",
      "notes2name": "",
      "notes3name": "",
      "notes4name": "",
      "notes1": {
        "value": ""
      },
      "notes": {
        "value": ""
      },
      "notes4": {
        "value": ""
      },
      "notes3": {
        "value": ""
      },
      "notes2": {
        "value": ""
      }

p4535992 commented:
> @p4535992 I sent it. Gender, Age, Height, Weight, Eyes, Skin and Hair are also wipe.
> 
> If I compare both files, this raws arn't keep (maybe more) :
> 
> ```
>   "gender": "",
>   "age": "",
>   "height": "",
>   "weight": "",
>   "eyes": "",
>   "skin": "",
>   "hair": "",
>   "notes1name": "",
>   "notes2name": "",
>   "notes3name": "",
>   "notes4name": "",
>   "notes1": {
>     "value": ""
>   },
>   "notes": {
>     "value": ""
>   },
>   "notes4": {
>     "value": ""
>   },
>   "notes3": {
>     "value": ""
>   },
>   "notes2": {
>     "value": ""
>   }
> ```

is is possible you run the macro on a 2.1.X world instead a 2.0.3 ?

p4535992 commented:
NEW BETA: https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.8/module.json

nivthefox commented:
Just in time to do it again for v11!

CaosFR commented:
> is possible you run the macro on a 2.1.X world instead a 2.0.3 ?

If I run create a export of my 2.0.3, import it then run the script it's dont work. 
But if I run your script in my 2.0.3 befoe export it's ok :)

sdenec commented:
You guys are great! I'm really happy you put in this much effort to keep my little module alive and improve upon my initial stuff! Just give me the green light when it's ready to merge. Also what official version number would be appropiate?
Lets hope v11 is rather tame and isn't again changing this much stuff around that affects sheets -.-

p4535992 commented:
There is only this issue i cannot solve @sdenec  maybe you can do it with soe css trick ?

The module setting _"Use classic item controls in list view."_ when enabled it shows the actions correctly but the section headers are "not synchronized at the column level" should be fixed with CSS ...

  ![img](https://i.imgur.com/UoIz6Y1.png)

CaosFR commented:
After some test, your script work only for PC, NPC are still wipe :(

RaylaTheBrave commented:
Am I the only one who's having this graphical bug?
![bug](https://user-images.githubusercontent.com/123700650/214965273-ada80c3c-4425-4cbe-b4ac-1b954cfeb2c9.PNG)

p4535992 commented:
> 

I think that is the module  [Character Actions 5e](https://github.com/ElfFriend-DnD/foundryvtt-dnd5eCharacterActions)  ? there is probably some label to update

RaylaTheBrave commented:
Yes I am using that module, then sorry for the confusion by my part

CaosFR commented:
@p4535992 I have modify your script for keep NPC journal, hope it's help 

```
//Update old tidy5e NPC Traits data
game.actors.updateAll((actor)=>{
    return {
        "flags.tidy5e-sheet":{
            trait:actor.system.details?.trait ?? actor.flags["tidy5e-sheet"]?.trait ?? "",
            ideal:actor.system.details?.ideal ?? actor.flags["tidy5e-sheet"]?.ideal ?? "",
            bond:actor.system.details?.bond ?? actor.flags["tidy5e-sheet"]?.bond ?? "",
            flaw:actor.system.details?.flaw ?? actor.flags["tidy5e-sheet"]?.flaw ?? "",
        notes: {
                "value": actor.system.details?.notes?.value ?? actor.flags["tidy5e-sheet"]?.notes?.value ?? ""
        },
               notes1:{
               "name": actor.system.details?.notes1name ?? actor.flags["tidy5e-sheet"]?.notes1?.name ?? "",
               "value": actor.system.details?.notes1?.value ?? actor.flags["tidy5e-sheet"]?.notes1?.value ?? ""
        },
           notes2:{
               "name": actor.system.details?.notes2name ?? actor.flags["tidy5e-sheet"]?.notes2?.name ?? "",
               "value": actor.system.details?.notes2?.value  ?? actor.flags["tidy5e-sheet"]?.notes2?.value ?? ""
        },
          notes3:{
               "name": actor.system.details?.notes3name ?? actor.flags["tidy5e-sheet"]?.notes3?.name ?? "",
               "value": actor.system.details?.notes3?.value ?? actor.flags["tidy5e-sheet"]?.notes3?.value ?? ""
        },
               notes4:{
               "name": actor.system.details?.notes4name ?? actor.flags["tidy5e-sheet"]?.notes4?.name ?? "",
               "value": actor.system.details?.notes4?.value ?? actor.flags["tidy5e-sheet"]?.notes4?.value ?? ""
        }
        }
    }
}, (actor)=>{
	return actor.type === 'npc';
});
```

NekroDarkmoon commented:
> There is only this issue i cannot solve @sdenec maybe you can do it with some css trick ?
> 
> The module setting _"Use classic item controls in list view."_ when enabled it shows the actions correctly but the section headers are "not synchronized at the column level" maybe we can fix with some CSS ?
> 
> ![img](https://camo.githubusercontent.com/d810e3f09f388159fafc83e86fab23bc0ca407b005121650e99084e99b7b498d/68747470733a2f2f692e696d6775722e636f6d2f556f497a3659312e706e67)

So I was looking to help with this, but like I'd really like to know why grid wasn't used for this 😅

NekroDarkmoon commented:
I believe the width of the inventory or the page somewhere was changed and as such all the static width values for the headers no longer match the original.

p4535992 commented:
Like you said i was trying to follow the system template, but maybe is more simple to rewrite the template to use a grid...

NekroDarkmoon commented:
```css
.tidy5e.sheet.actor .items-list .items-header-labels .items-header-controls {
  width: 85px;
}
```

NekroDarkmoon commented:
Here's a simple change to fix it, I'm not sure what the setting toggle is and if it breaks in the "non-classic list menu" but this does bring it back to it's original style.

![image](https://user-images.githubusercontent.com/49667684/215097932-d201eb4a-f148-44a0-b6a1-2815ba09ac68.png)

GambetTV commented:
This is looking real good! Just a couple observations from a user point of view: 

1. It looks like Favorited items are in random order rather than alphabetical.
2. I am unclear on why a second flag shows up on favorited items. When you favorite an item you have to click on a little flag symbol which is grayed out, but then highlights when active, and now a second flag symbol pops up to the left of the item's weight. Just feels like unnecessary clutter.
3. There needs to be a way to disable the Duplicate button. People are gonna accidentally hit this, and it doesn't even prompt you to make sure you meant to duplicate an item. Also I don't even know what the use case is for duplicating Spells or Class/Race features. At this very least this button should go invisible when the character sheet is locked, but I'd vastly prefer an option in Tidy5e Settings where I can simply disable this completely for Players and the GM so it doesn't even show up.

That's really it! This is a great effort, and I thank you all for keeping this module alive!

sdenec commented:
> Like you said i was trying to follow the system template, but maybe is more simple to rewrite the template to use a grid...

I tried using grid but I have to say I'm not fully comfortable with it, yet (I know way behind on it) so I used flex. I'm certain grid could make it cleaner and if anyone wants to tackle it - go for it!

NekroDarkmoon commented:
For now, a min-width set on a few things would fix it. But deffo a hacky way to about it. Switching to grid would be boon to the entire thing and cut down on many lines of unnecessary css.

p4535992 commented:
> * It looks like Favorited items are in random order rather than alphabetical. (fixed )
> * I am unclear on why a second flag shows up on favorited items. When you favorite an item you have to click on a little flag symbol which is grayed out, but then highlights when active, and now a second flag symbol pops up to the left of the item's weight. Just feels like unnecessary clutter. (i think is more clear, but you need for when the classic ontrols menu is disabled
> * There needs to be a way to disable the Duplicate button. People are gonna accidentally hit this, and it doesn't even prompt you to make sure you meant to duplicate an item. Also I don't even know what the use case is for duplicating Spells or Class/Race features. At this very least this button should go invisible when the character sheet is locked, but I'd vastly prefer an option in Tidy5e Settings where I can simply disable this completely for Players and the GM so it doesn't even show up. (for now i hide the duplicate functions togheter with the delete functions with the toogle lock let'see if we need more customization for this.

p4535992 commented:
I think i fixed  everything here the new (and i hope final) BETA: https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.9/module.json

p4535992 commented:
as i was saying final beta release (maybe ?): https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.10/module.json

p4535992 commented:
NEW BETA : https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.11/module.json

DerJJ commented:
The new buttons have a different alignment comparing cantrips to leveled spells, is this normal?
A Sorcerers Spellbook in the 0.8.11 beta.

![sb8 11](https://user-images.githubusercontent.com/437987/215266089-0bbe74f3-2181-4aec-93be-7d8edb9918c7.png)

szefo09 commented:
I suggest removing the migration on Item Sheet render, No point and will cause issues whenever someone tries to open Item Sheet.
https://github.com/p4535992/tidy5e-sheet/blob/499a51388469d2b42e450b6331ec3cdcea8f7877/src/scripts/tidy5e-item.js#L38

p4535992 commented:
New Beta: https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.12/module.json

MaxPat931 commented:
This beta looks good to me, not sure if there is anything specific you are interested in testing
The only feedback I have is in regards to the settings, there should be some text on the left of the Sheet Settings button just like "Tidy5e sheet settings - configure the basic features of the sheet"
and the use of the coins icon for the Reset button is a little strange
![image](https://user-images.githubusercontent.com/86370342/215301728-89a421c0-a246-4aa1-aba4-0bf484809191.png)

Also, I know this work is focused on the Actor sheet, but it seems this issue on the Advancements tab of Class/Subclass/Backgrounds has returned https://github.com/sdenec/tidy5e-sheet/issues/587
![image](https://user-images.githubusercontent.com/86370342/215302015-18a0e1f5-e56b-4b33-ad66-c6ea8f0ab4a8.png)

p4535992 commented:
New Beta: https://github.com/p4535992/tidy5e-sheet/releases/download/v0.8.13/module.json

Fixed: icons on advancement
Fixed: icon on reset settigns
Fixed: Middle Click edit item event

p4535992 commented:
@sdenec I think it is ready or at least it seems stable, i prepared the module.json with version 0.8.14

p4535992 commented:
highlight when you do the release on discord that if someone still needs to make the transition from 2.0.3 to 2.1.X they should use the macro on the readme.md BEFORE the migration

cs96and commented:
> highlight when you do the release on discord that if someone still needs to make the transition from 2.0.3 to 2.1.X they should use the macro on the readme.md BEFORE the migration

Probably worth a Reddit post too

sdenec commented:
Could one of you post to reddit - if needed? I don't use it^^

cs96and commented:
This doesn't look right in the migration macro...

`maxPreparedSpells: actor.system.details?.maxPreparedSpells ?? actor.flags["maxPreparedSpells"]?.hair ?? ""`

It's using `hair` at the end.

MaxPat931 commented:
Sorry but I am still seeing the HP Max editing #647  not work in the latest beta, when removing an override

https://user-images.githubusercontent.com/86370342/215358520-0f335476-bd3d-49cc-8a0f-4b83e3392b89.mp4

Here I delete the 32 which was set as the Max HP, when I click save the first time it correctly resets based on the HP advancements this character has set, then clicking save again resets it to a 0 override and saves the HP and closes the dialog

cs96and commented:
> This doesn't look right in the migration macro...
> 
> `maxPreparedSpells: actor.system.details?.maxPreparedSpells ?? actor.flags["maxPreparedSpells"]?.hair ?? ""`
> 
> It's using `hair` at the end.
> 
> This is from README.md

@p4535992 Looks like you fixed this, but you missed a `?`...
```js
maxPreparedSpells: actor.system.details?.maxPreparedSpells ?? actor.flags["tidy5e-sheet"].maxPreparedSpells ?? "", 
```
should be...
```js
maxPreparedSpells: actor.system.details?.maxPreparedSpells ?? actor.flags["tidy5e-sheet"]?.maxPreparedSpells ?? "", 
```
